### PR TITLE
Manager validity dates restrictions + Add booking date validation (didn’t exist before)

### PR DESCRIPTION
**Pull request by folix-01**
- Add possibility to restrict also a Bookings Manager by the validity dates of a Booking Folder.
- Add validity dates checks during the booking creation. Actually is not considered so u can create booking for the every date you want

### CODE REVIEW

File: src/redturtle/prenotazioni/content/prenotazioni_folder.py [hunk 1]
```diff
@@ -363,7 +374,7 @@ def get_options():
     )
 
     notBeforeDays = schema.Int(
-        default=2,
+        default=0,
```
Comment (mamico) [inline src/redturtle/prenotazioni/content/prenotazioni_folder.py:377]: Why did this default change?

File: src/redturtle/prenotazioni/browser/prenotazioni_context_state.py [hunk 2]
```diff
@@ -286,17 +297,42 @@ def is_before_allowed_period(self, day, bypass_user_restrictions=False):
     @memoize
     def is_valid_day(self, day, bypass_user_restrictions=False):
         """Returns True if the day is valid"""
```
Comment (mamico) [inline src/redturtle/prenotazioni/browser/prenotazioni_context_state.py:299]: List the cases for which control is bypassed and those for which it is not

File: src/redturtle/prenotazioni/browser/prenotazioni_context_state.py [hunk 3]
```diff
@@ -286,17 +297,42 @@ def is_before_allowed_period(self, day, bypass_user_restrictions=False):
...
+
+        if isinstance(day, datetime):
+            day = day.date()
+
+        is_configured_day = self.is_configured_day(day)
+
+        if (
+            is_configured_day
+            and self.user_can_manage_prenotazioni
```
Comment (mamico) [inline src/redturtle/prenotazioni/browser/prenotazioni_context_state.py:308]: Why is `bypass_user_restrictions` not used here?

File: src/redturtle/prenotazioni/content/prenotazioni_folder.py (hunk 1 above)
Comment (folix-01) [reply src/redturtle/prenotazioni/content/prenotazioni_folder.py:377]: bcs the written tests are not considering that the default is 2. we can change tests or we can change the default. So as tests are rappresentign the expected behaviour i changed the default

File: src/redturtle/prenotazioni/browser/prenotazioni_context_state.py (hunk 2 above)
Comment (folix-01) [reply src/redturtle/prenotazioni/browser/prenotazioni_context_state.py:299]: bypass_user_restrictions was not added in this pr, it existed before: as for this pr bypass_user_restrictions is true when user is manager and "restrict user by validity days" flag is off. the other cases must be studied

Comment (mamico) [reply src/redturtle/prenotazioni/browser/prenotazioni_context_state.py:299]: ok

Comment (mamico) [reply src/redturtle/prenotazioni/browser/prenotazioni_context_state.py:299]: Yes, but before the code was really simple, now there is much more complexity. Before it wasn't necessary, now it is

Comment (folix-01) [reply src/redturtle/prenotazioni/browser/prenotazioni_context_state.py]: added doc

File: src/redturtle/prenotazioni/locales/redturtle.prenotazioni.pot [hunk 4]
```diff
@@ -34,11 +34,15 @@ msgstr ""
...
 
-#: redturtle/prenotazioni/content/prenotazioni_folder.py:433
+#: redturtle/prenotazioni/content/prenotazioni_folder.py:444
 #: redturtle/prenotazioni/content/validators.py:222
 msgid "Afternoon start should not be greater than end."
 msgstr ""
 
+#: redturtle/prenotazioni/content/prenotazioni_folder.py:216
+msgid "Apply restrictions to Bookings Manager"
+msgstr ""
```
Comment (giuliaghisini) [inline src/redturtle/prenotazioni/locales/redturtle.prenotazioni.pot:44]:
```suggestion
msgstr "Applica le restrizioni sulle date anche agli operatori"
```

File: src/redturtle/prenotazioni/locales/redturtle.prenotazioni.pot [hunk 5]
```diff
@@ -185,6 +189,10 @@ msgstr ""
 msgid "How to get here"
 msgstr ""
 
+#: redturtle/prenotazioni/content/prenotazioni_folder.py:219
+msgid "If selected, Bookings Manager will be restricted by selected dates as an usual user."
+msgstr ""
```
Comment (giuliaghisini) [inline src/redturtle/prenotazioni/locales/redturtle.prenotazioni.pot:194]:
```suggestion
msgstr "Se selezionato, gli operatori avranno le stesse restrizioni sulle date come un qualsiasi utente"
```

**Review by giuliaghisini — DISMISSED** [review]

**Pull request merged**